### PR TITLE
fix service invite flow and skip tests for now

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

The service invite was working locally and failing on staging.

The reason was determined to be that the service invite workflow was trying to write to redis in the back end, and then read from redis on the front end, assuming that the redis instance was shared.  The redis instance actually is shared in local development.  But it is not shared on staging.

So the solution is to write an API that returns the value in redis containing all the service invite information.  This ultimately resulted in the following work being necessary:

1. Write the API
2. Modify some tests to mock where the API call happens so they don't blow up if they are not even interested in service invites
3. Remove some redis mocks from tests
4. Try to clean up the workflow.  The old deprecated way where we were not sending invited users off to login.gov was still passing its tests even though it did not work in real life anymore.  On the other hand, the new login.gov way fails all its (old) tests even though it appears to work in real life now.
5. In the interest of time, skip the failing tests so we can get a fix to users.  There is a follow up ticket to fix all the tests which will be the next thing I work on.

NOTA BENE:

There is also some weirdness here.  See the service_api_client.py.  For some reason the method belonging to ServiceAPIClient could not called directly from verify.py (pycharm said the method didn't belong to the class), so I had to make the wrapper method retrieve_service_invite_data at the bottom of that file.

## Security Considerations

N/A

### CODE REVIEW

File: app/main/views/verify.py [hunk 1]
```diff
@@ -69,83 +67,50 @@ def activate_user(user_id):
...
-    )
-    # END DEBUG
+    except BaseException:  # noqa
+        # We will hit an exception if we can't find invite data,
+        # but that will be the normal sign in use case
+        login_gov_invite_data = None
     if login_gov_invite_data:
-        login_gov_invite_data = json.loads(login_gov_invite_data.decode("utf8"))
+        # Not clear why we have to call json.loads twice, but we do
+        login_gov_invite_data = json.loads(login_gov_invite_data)
```
Comment (ccostino) [inline app/main/views/verify.py:80]: That definitely seems odd, unless it's a JSON string within a JSON string?

**Review by ccostino — COMMENTED** [review]
Thanks, @terrazoon!  I think this all looks good to go once the API side is in.

I had a quick thought on the JSON piece you called out, too.

File: app/main/views/verify.py (hunk 1 above)
Comment (terrazoon) [reply app/main/views/verify.py:80]: Oh, that comment is out of date.  When I was sending data back from api, I was calling json.dumps(), but apparently there is an implicit json.dumps() already happening.  I have removed that from the API, so only one json.loads() is needed here.  I will remove the comment.

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon!

Noting here that the tests that are marked for skip with this change will be unmarked soon as the remaining work is finished.

**Pull request merged**